autorest/codegen/serializers/__init__.py

```diff
@@ -54,125 +54,129 @@ def serialize(self, code_model: CodeModel) -> None:
+    def _serialize_and_write_operations_folder(self, code_model: CodeModel, env: Environment) -> None:
         namespace_path = Path(*(code_model.namespace.split(".")))
         # write sync operations init file
         operations_init_serializer = OperationsInitSerializer(code_model=code_model, env=env, async_mode=False)
-        operations_init_serializer.serialize()
         self._autorestapi.write_file(
             namespace_path / Path(f"operations") / Path("__init__.py"),
-            operations_init_serializer.operations_init_file
+            operations_init_serializer.serialize()
         )
 
         # write async operations init file
         if not code_model.options['no_async']:
             operations_async_init_serializer = OperationsInitSerializer(code_model=code_model, env=env, async_mode=True)
-            operations_async_init_serializer.serialize()
             self._autorestapi.write_file(
                 namespace_path / Path("aio") / Path(f"operations_async") / Path("__init__.py"),
-                operations_async_init_serializer.operations_init_file
+                operations_async_init_serializer.serialize()
             )
 
         for operation_group in code_model.operation_groups:
             # write sync operation group and operation files
             operation_group_serializer = OperationGroupSerializer(
                 code_model=code_model, env=env, operation_group=operation_group, async_mode=False
             )
-            operation_group_serializer.serialize()
             self._autorestapi.write_file(
                 namespace_path / Path(f"operations") / Path(operation_group_serializer.filename()),
-                operation_group_serializer.operation_group_file
+                operation_group_serializer.serialize()
             )
 
             if not code_model.options['no_async']:
                 # write async operation group and operation files
                 operation_group_async_serializer = OperationGroupSerializer(
                     code_model=code_model, env=env, operation_group=operation_group, async_mode=True
                 )
-                operation_group_async_serializer.serialize()
                 self._autorestapi.write_file(
                     (namespace_path / Path("aio") / Path(f"operations_async") /
                     Path(operation_group_async_serializer.filename())),
-                    operation_group_async_serializer.operation_group_file
+                    operation_group_async_serializer.serialize()
                 )
 
-    def _serialize_and_write_top_level_folder(self, code_model, env):
+    def _serialize_and_write_top_level_folder(self, code_model: CodeModel, env: Environment) -> None:
         general_serializer = GeneralSerializer(code_model=code_model, env=env, async_mode=False)
-        general_serializer.serialize()
 
         namespace_parts = code_model.namespace.split(".")
-        namespace_path = None
+        namespace_path: Path = Path()
```

Comment:
namespace_path: Path = Path(".")

Reply:
Passing in namespace now to every function and checking for no-namespace-folders flag